frontend/public/components/import-yaml-results.tsx

```diff
@@ -0,0 +1,236 @@
+  TableGridBreakpoint,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+
+import { ResourceLink } from './utils';
+import { createRequests } from './edit-yaml';
+import { K8sKind, referenceFor } from '../module/k8s';
+
+/**
+ * Without this prop our current TS types fail to match and require a `translate` prop to be added. PF suggests we
+ * update our types, but that causes other issues. This will have to do as a workaround for now.
+ *
+ * This is the best that I can find relates to this value:
+ * https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3423b4fc3e3da09f8acc386bc2fee6fb8f5e0880/types/react/index.d.ts#L1763
+ */
+const reactPropFix = {
+  translate: 'no',
+};
+
+const ImportYAMLPageStatus: React.FC<ImportYAMLStatusProps> = ({ errors, inFlight }) => {
+  const { t } = useTranslation();
+  let StatusBlock: React.ReactNode;
+
+  if (inFlight) {
+    StatusBlock = (
+      <>
+        <Spinner size="lg" />
+        <h2>{t('public~Creating resources...')}</h2>
+      </>
+    );
+  } else if (!inFlight && !errors) {
+    StatusBlock = (
+      <>
+        <GreenCheckCircleIcon size="lg" />
+        <h2>{t('public~Resources successfully created')}</h2>
+      </>
+    );
+  } else {
+    StatusBlock = (
+      <>
+        <YellowExclamationTriangleIcon size="lg" />
+        <h2>{t('public~One or more resources failed to be created')}</h2>
+      </>
+    );
+  }
+  return <div className="co-import-yaml-status text-center">{StatusBlock}</div>;
+};
+
+const ImportYAMLResourceStatus: React.FC<ImportYAMLResourceStatusProps> = ({
+  creating,
+  error,
+  message,
+}) => {
+  let StatusIcon;
+  if (creating) {
+    StatusIcon = <Spinner size="sm" className="co-icon-space-r" />;
+  } else if (error) {
+    StatusIcon = <RedExclamationCircleIcon className="co-icon-space-r" size="sm" />;
+  } else {
+    StatusIcon = <GreenCheckCircleIcon className="co-icon-space-r" size="sm" />;
+  }
+  return (
+    <span>
+      {StatusIcon}
+      {message}
+    </span>
+  );
+};
+
+export const ImportYAMLResults: React.FC<ImportYAMLResultsProps> = ({
+  importResources,
+  models,
+  retryFailed,
+  showResults,
+}) => {
+  const { t } = useTranslation();
+  const [importStatus, setImportStatus] = React.useState<ImportYAMLResourceStatus[]>(
+    importResources.map((resource, index) => ({
+      id: index,
+      creating: true,
+      message: t('public~Creating'),
+    })),
+  );
+  const [inFlight, setInFlight] = React.useState(true);
+  const [errors, setErrors] = React.useState(false);
+  React.useEffect(() => {
+    const requests = createRequests(importResources, models);
+    Promise.allSettled(requests).then(() => {
+      setInFlight(false);
+    });
+    requests.forEach((resourceRequest, index) => {
+      resourceRequest
+        .then(() =>
+          setImportStatus((prevResources) => {
+            const nextResources = [...prevResources];
+            nextResources[index] = {
+              id: index,
```

Comment:
If `id` is just the index in the array, why do we need `id` at all? Can't we just use the index, which we should already know?

Reply:
@spadgett i was using the index in the retry method to easily index the filtered list of error objects, but maybe i can adjust that to remove the filter , and just use map so I have an index.

Reply:
We should be able to use `Array.reduce`, something like

```ts
const failedResourceObjects = importStatus.reduce((resource, arr, i) => {
  if (!resource.error) {
    return arr;
  }
  return [...arr, importResources[i]];
}, []);
```